packages/material-ui-styles/src/styled/styled.js

```diff
@@ -82,6 +82,18 @@ export default function styled(Component) {
       const classes = useStyles(props);
       const className = clsx(classes.root, classNameProp);
 
+      if (props.css) {
+        if (process.env.NODE_ENV !== 'production') {
```

Comment:
We can swap the two conditions.

Reply:
We probably want to warn only once. It will be very annoying otherwise.

Reply:
Done